core/amber/src/main/resources/python_udf/texera_udf_operator_base.py

```diff
@@ -167,3 +167,35 @@ def report_matrix(self, y_test, y_pred, *args):
         for index, row in list(matrix.iterrows())[::-1]:
             if index != 1:
                 self._result_tuples.append(row)
+
+class TexeraBlockingUnsupervisedTrainerOperator(TexeraUDFOperator):
+
+    def __init__(self):
+        super().__init__()
+        self._data = []
+        self._result_tuples: List = []
+        self._train_args = dict()
+
+    def input_exhausted(self, *args):
+        model = self.train(self._data, **self._train_args)
+        self.report(model)
+
+    def accept(self, row: pandas.Series, nth_child: int = 0) -> None:
+        self._data.append(row[0])
+
+    def has_next(self) -> bool:
+        return bool(self._result_tuples)
+
+    def next(self) -> pandas.Series:
+        return self._result_tuples.pop()
```

Comment:
I just realized this would result in a wrong order (FILO), but we want FIFO. so change it to 
```suggestion
        return self._result_tuples.pop(0)
```
so that it removes from head as a queue.

Reply:
I will do the same for supervised trainer in future PRs.